algoliasearch/index.go

```diff
@@ -433,24 +382,48 @@ func (i *Index) ClearSynonyms(forwardToSlaves bool) (interface{}, error) {
 // with `forwardToSlaves`
 // An error is returned if the underlying HTTP call does not yield a 200
 // status code.
-func (i *Index) BatchSynonyms(synonyms []interface{}, replaceExistingSynonyms, forwardToSlaves bool) (interface{}, error) {
-	params := i.client.EncodeParams(map[string]interface{}{
-		"forwardToSlaves":         strconv.FormatBool(forwardToSlaves),
-		"replaceExistingSynonyms": strconv.FormatBool(replaceExistingSynonyms),
-	})
+func (i *Index) BatchSynonyms(synonyms []Synonym, replaceExistingSynonyms, forwardToSlaves bool) (res UpdateTaskRes, err error) {
+	if err = checkSynonyms(synonyms); err != nil {
+		return
+	}
+
+	params := map[string]bool{
+		"replaceExistingSynonyms": replaceExistingSynonyms,
+		"forwardToSlaves":         forwardToSlaves,
+	}
 
-	return i.client.transport.request("POST", "/1/indexes/"+i.nameEncoded+"/synonyms/batch?"+params, synonyms, write)
+	path := i.route + "/synonyms/batch?" + i.client.encodeParams(params)
+	err = i.client.request(&res, "POST", path, synonyms, write)
+	return
 }
 
-// SaveSynonym adds the given `synonym` to be identified `objectID`.
-// This addition can be forwarded to the index slaves using `forwardToSlaves`.
-// An error is returned if the underlying HTTP call does not yield a 200
-// status code.
-func (i *Index) SaveSynonym(objectID string, synonym interface{}, forwardToSlaves bool) (interface{}, error) {
-	encodedID := i.client.EncodeParams(objectID)
-	params := i.client.EncodeParams(map[string]interface{}{
-		"forwardToSlaves": strconv.FormatBool(forwardToSlaves),
-	})
+// Browse returns `hitsPerPage` results from the `page` page.
+// Deprecated: Use `BrowseFrom` or `BrowseAll` instead.
+func (i *Index) Browse(params map[string]interface{}) (res BrowseRes, err error) {
+	if err = checkQuery(params); err != nil {
+		return
+	}
+
+	path := i.route + "/browse?" + i.client.encodeParams(params)
+	err = i.client.request(&res, "GET", path, nil, read)
+	return
+}
+
+func (i *Index) BrowseAll(params map[string]interface{}) (IndexIterator, error) {
+	return NewIndexIterator(i, params)
+}
+
+// Search performs a search query according to the `query` search query and the
+// given `params` parameters.
+func (i *Index) Search(query string, params map[string]interface{}) (res QueryRes, err error) {
+	copy := duplicateMap(params)
+	copy["query"] = query
+
+	req := map[string]interface{}{
+		"params": i.client.encodeParams(params),
+	}
 
-	return i.client.transport.request("PUT", "/1/indexes/"+i.nameEncoded+"/synonyms/"+encodedID+"?"+params, synonym, write)
+	path := i.route + "/query"
+	err = i.client.request(&res, "POST", path, req, search)
```

Comment:
Right. Plus I was using `params` instead of copy in the `encodeParams`. Everything's fixed.